### ISSUE TITLE
Refactor `probnum.diffeq` to resolve the pylint messages

### ISSUE THREAD

**Issue by nathanaelbosch** [probabilistic-numerics/probnum#156]
At the time of writing, `probnum.diffeq` triggers the following `pylint` messages:
- `protected-access`
- `abstract-class-instantiated`
- `too-many-locals`
- `too-few-public-methods`
- `too-many-arguments`
- `unused-argument`

__Goal:__
Then, after fixing all errors/warnings, you should remove the exception from the corresponding line in `./tox.ini`, such that this message will be tracked in the future.

__To work on this issue__ you can check for these messages individually by calling:
```
pylint src/probnum/diffeq --disable=all --enable=<message>
```
Then, after fixing all errors/warnings, you should remove the exception from the corresponding line in `./tox.ini`, such that this message will be tracked in the future.

**Comment by pnkraemer** [comment]
I am back now. Shall we make a game plan in the next days?

**Comment by nathanaelbosch** [comment]
Yes sure! I'd propose we meet and and go through the error messages together, but let's talk about it in person tomorrow or during the next days.
